lib/fluent/plugin/out_aws-elasticsearch-service.rb

```diff
@@ -63,10 +64,14 @@ def credentials(opts)
           credentials = Aws::Credentials.new opts[:access_key_id], opts[:secret_access_key]
         else
           if opts[:assume_role_arn].nil?
-            credentials = Aws::SharedCredentials.new({
-                             retries: 2
-                          }).credentials
-            credentials ||= Aws::InstanceProfileCredentials.new.credentials
+            if opts[:ecs_container_credentials_relative_uri].nil?
+              credentials = Aws::SharedCredentials.new({retries: 2}).credentials
```

Comment:
Don't you want to do credentials ||= Aws::ECSCredentials.new.credentials here as well, in case the config is not specified?